app/shop.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def __str__(self) -> str:
+        return f"{self.name}"
+
+    def costs_for_trip_to_shop(self,
+                               customer: Any,
+                               fuel_price: float) -> float:
+        return self._calculate_fuel_price(
+            customer,
+            fuel_price
+        ) + self.calculate_cart_price(customer)
+
+    def receipts(self, customer: Any) -> str:
+        total_cost = self.calculate_cart_price(customer)
+        purchase_info = "You have bought:\n"
+        for product, quantity in customer.product_cart.items():
+            if product in self.products:
+                price = self.products[product]
+                cost = float(price * quantity)
+                cost = int(cost) if cost.is_integer() else cost
+                purchase_info += f"{quantity} {product}s for "
+                purchase_info += f"{cost} dollars\n"
+        purchase_info += f"Total cost is {total_cost} dollars\nSee you again!"
+        return purchase_info
+
+    def calculate_cart_price(self, customer: Any) -> float:
+        cart_price = 0
+        for product, quantity in customer.product_cart.items():
+            cart_price += self.products[product] * quantity
+        return cart_price
+
+    def _calculate_fuel_price(self,
+                              customer: Any,
+                              fuel_price: float) -> float:
+        fuel_needed = self._calculate_fuel_needed(
+            customer.location,
+            customer.car.fuel_consumption
+        )
+        return round((fuel_needed * fuel_price) * 2, 2)
+
+    def _calculate_fuel_needed(self,
+                               location: list,
+                               fuel_consumption: float
+                               ) -> float:
+        return (self._calculate_distance(location) * fuel_consumption) / 100
+
+    def _calculate_distance(self, customer_location: list) -> float:
+        distance_squared = ((self.location[0] - customer_location[0]) ** 2
+                            + (self.location[1] - customer_location[1]) ** 2)

```

Comment:
```suggestion
        distance_squared = ((self.location[0] - customer_location[0]) ** 2
        distance_squared *= 2
```
We just can multiple by 2
For example: if from our home to shop distance will be 100meters is it possible that from shop to home distance will be bigger or smaller?

Reply:
((self.location[0] - customer_location[0]) ** 2 + (self.location[1] - customer_location[1]) ** 2)

this is formula for 2d distance calculate from A to B 
distance = math.sqrt((point_B[0] - point_A[0])**2 + (point_B[1] - point_A[1])**2)